### PR TITLE
fix: weird behaviour of dates in base node banned peers

### PR DESCRIPTION
**Pull request by mrnaveira**
Description
---
* Changed the peer display format to output `offline_at` and `banned_until` as local time instead of UTC, fixing the “time in the past” issue
* Improved handling and display of permanent bans (when invoking `ban-peer` with no params). Note that this makes _existing_ permabans still display weird times, changes affect only new permabans
* Scope of the changes does seem to only affect log messages in other applications

Motivation and Context
---
There is some weird behaviour regarding display of date times on banned peers.
- Times are shown in UTC but without the timezone, so if the user is in UTC+N the ban end time may look like it's "in the past" for some values
- The default ban time is permanent, and this makes the "banned until" display the max datetime.

How Has This Been Tested?
---
Manually tested on a base node, by banning peers with different values

### CODE REVIEW

**Review by sdbondi — DISMISSED** [review]
Looks good!

File: comms/core/src/utils/datetime.rs [hunk 1]
```diff
@@ -48,6 +46,15 @@ pub fn format_duration(duration: Duration) -> String {
     }
 }
 
+pub fn format_local_datetime(datetime: &NaiveDateTime) -> String {
+    let local_datetime = DateTime::<Local>::from_utc(*datetime, Local::now().offset().to_owned());
+    format!("{}", local_datetime.format("%Y-%m-%d %H:%M:%S"))
```
Comment (sdbondi) [inline comms/core/src/utils/datetime.rs]: nit:
```suggestion
    local_datetime.format("%Y-%m-%d %H:%M:%S")).to_string()
```

Comment (mrnaveira) [reply comms/core/src/utils/datetime.rs]: good one, thanks

**Review by sdbondi — APPROVED** [review]

**Review by SWvheerden — APPROVED** [review]
Looks good to me
Spent about 5 mins on this.
Tested and run locally

**Pull request merged**